lib/pat.c

```diff
@@ -822,7 +822,7 @@ grn_pat_wal_add_entry_format_details(grn_ctx *ctx,
     grn_text_printf(ctx, details, "key-size:%u ", data->key_size);
   }
   if (used->key_offset) {
-    grn_text_printf(ctx, details, "key-offset:%u ", data->key_offset);
+    grn_text_printf(ctx, details, "key-offset:%lu ", data->key_offset);
```

Comment:
We should use `GRN_FMT_INT64U`:

```suggestion
    grn_text_printf(ctx, details, "key-offset:%" GRN_FMT_INT64U " ", data->key_offset);
```